crates/duckdb/src/arrow_batch.rs

```diff
@@ -7,12 +9,24 @@ use super::{
 #[must_use = "Arrow is lazy and will do nothing unless consumed"]
 pub struct Arrow<'stmt> {
     pub(crate) stmt: Option<&'stmt Statement<'stmt>>,
+    pub(crate) schema: Option<SchemaRef>,
```

Comment:
It might make it clearer if this wasn't just `schema` but an enum that contained the schema, so it was more obvious what it was for, i.e.:

```rust
enum Mode {
  Streaming(SchemaRef)
  Batch
}
```

or to match the underlying terminology:

```rust
enum Mode {
  StreamingStep(SchemaRef)
  Step
}
```

Then this becomes a:
```rust
pub(crate) mode: Mode
```